src/views/ActivityAppFeed.vue

```diff
@@ -156,7 +156,10 @@ async function loadActivities() {
 		)
 		allActivities.value.push(...response.data.ocs.data.map((raw) => new ActivityModel(raw)))
 		lastActivityLoaded.value = response.headers['x-activity-last-given']
-		hasMoreActivites.value = true
+		// If less than the hardcoded limit, there are no more activities
+		if (response.data.ocs.data.length < 50) {
+			hasMoreActivites.value = false
```

Comment:
Possible improvement in the future: Check again after a timeout if there is something new.

Reply:
This completely breaks the activity stream and needs reverting!

Basically if you upload more than 50 files, it will be the end of your activity stream. In fact 3 uploads will be enough already. You ask the backend for 50 entries, but only get 48, as 3 of them are combined into 1 entry

I see it's also broken before with the loading spinner you tried to fix, but that is another problem. Checking the result size is not the solution. Whether or not you reached the end of the activity stream is indicated by the HTTP status code 304:
https://github.com/nextcloud/activity/blob/master/docs/endpoint-v2.md#http-status